src/cmd/ksh93/tests/echo.exp

```diff
@@ -0,0 +1,36 @@
+# Test for echo builtin
+set pid [spawn $ksh]
+expect_prompt
+
+# ==========
+log_test_entry
+send "echo -n hello world\r"
+expect "\r\nhello world" {
```

Comment:
This doesn't do what you think it does. Hint: Does this pass or fail if you change the command to "echo hello world\r"?

There are a couple of ways to fix this. I would fix it by replacing these two statements with this:

```
send "echo -n hello world; echo EOT\r"
expect "\r\nhello worldEOT\r\n" {
```

FWIW, I frequently make this mistake. Writing robust expect based tests is hard. Which is why I will often deliberately change the test to violate the condition being tested to confirm the expect pattern does what I'm expecting.